### PR TITLE
Document template fixes

### PR DESCRIPTION
**Pull request by ricopinazo**
### What changes were proposed in this pull request?

- [x] Add default document templates
- [x] Update vector API (on the server as well) to allow choosing between using the default template, a custom one, or nothing at all, for each of the three types of entities
- [x] Fix a bug causing subgraphs to allow containing the same node more than once
- [x] Fix a bug causing NaN float to panic when querying through GraphQL
- [x] Review public API to stick to temporal_props / constant_props naming convention

### Why are the changes needed?
Having default templates is a first step towards a smart search view on the generic UI

### Does this PR introduce any user-facing change? If yes is this documented?
Yes, and the docstrings were updated accordingly

### How was this patch tested?
- New tests were added to test the default templates in rust
- A new python test was added to double-check the default template is used if the user doesn't specify anything

### Are there any further changes required?
No

### CODE REVIEW

File: raphtory/src/db/graph/views/node_subgraph.rs [hunk 1]
```diff
@@ -60,6 +60,7 @@ impl<'graph, G: GraphViewOps<'graph>> NodeSubgraph<G> {
             nodes.collect()
         };
         nodes.sort();
+        nodes.dedup();
```
Comment (ricopinazo) [inline raphtory/src/db/graph/views/node_subgraph.rs:63]: add a test to confirm this

File: raphtory/src/python/graph/properties/constant_props.rs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 use crate::{
     core::Prop,
     db::api::properties::{
-        dyn_props::DynConstProperties, internal::PropertiesOps, ConstProperties,
+        dyn_props::DynConstProperties, internal::PropertiesOps, ConstantProperties,
```
Comment (ricopinazo) [inline raphtory/src/python/graph/properties/constant_props.rs:4]: rename this file name (and search for others)

File: raphtory/src/vectors/template.rs [hunk 3]
```diff
@@ -201,7 +204,10 @@ impl DocumentTemplate {
                 let mut env = Environment::new();
                 let template = build_template(&mut env, template);
                 match template.render(NodeTemplateContext::from(node)) {
-                    Ok(document) => Box::new(std::iter::once(document.into())),
+                    Ok(mut document) => {
+                        truncate(&mut document);
```
Comment (ricopinazo) [inline raphtory/src/vectors/template.rs:208]: mention this in the docs / try to make this configurable

Comment (ricopinazo) [reply raphtory/src/vectors/template.rs:208]: add helper funtion node.generate_document() to allow debugging

Comment (ricopinazo) [reply raphtory/src/vectors/template.rs:208]: ideally simply use multi documents for big documents

File: raphtory/src/vectors/template.rs [hunk 4]
```diff
@@ -274,14 +290,102 @@ impl<'graph, G: GraphViewOps<'graph>> From<EdgeView<G>> for EdgeTemplateContext
...
+{% endfor %}";
+
+pub const DEFAULT_GRAPH_TEMPLATE: &str = "Graph with the following properties:
+{% for (key, value) in constant_properties|items %}
+{{ key }}: {{ value }}
+{% endfor %}
+{% for (key, values) in temporal_properties|items %}
+{{ key }}:
+{% for (time, value) in values %}
+ - changed to {{ value }} at {{ time|datetimeformat }}
```
Comment (ricopinazo) [inline raphtory/src/vectors/template.rs:318]: allow choosing between numbers or timestamps in the API

**Review by miratepuffin — APPROVED** [review]
Pair reviewed with Pedro - tickets being made for spotted issues, but LGTM for now

**Pull request merged**